main.go

```diff
@@ -228,3 +129,60 @@ func normalizeAddr(s string) string {
 
 	return u.String()
 }
+
+
+// parseConfig is broken into its own function for testability
+func parseConfig() (*Configuration, error) {
+	config := &Configuration{}
+	if err := envconfig.Process("", config); err != nil {
+		return nil, err
+	}
+	if err := config.Validate(); err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+type Configuration struct {
+	// Required
+	SecurityUserName     string `envconfig:"security_user_name"`
+	SecurityUserPassword string `envconfig:"security_user_password"`
+	VaultToken           string `envconfig:"vault_token"`
+
+	// Optional
+	CredhubURL         string   `envconfig:"credhub_url"`
+	Port               string   `envconfig:"port" default:":8000"`
+	ServiceID          string   `envconfig:"service_id" default:"0654695e-0760-a1d4-1cad-5dd87b75ed99"`
+	VaultAddr          string   `envconfig:"vault_addr" default:"https://127.0.0.1:8200"`
+	VaultAdvertiseAddr string   `envconfig:"vault_advertise_addr"`
+	ServiceName        string   `envconfig:"service_name" default:"hashicorp-vault"`
+	ServiceDescription string   `envconfig:"service_description" default:"HashiCorp Vault Service Broker"`
+	PlanName           string   `envconfig:"plan_name" default:"shared"`
+	PlanDescription    string   `envconfig:"plan_description" default:"Secure access to Vault's storage and transit backends"`
+	ServiceTags        []string `envconfig:"service_tags"`
+	VaultRenew         bool     `envconfig:"vault_renew" default:"true"`
+}
+
+func (c *Configuration) Validate() error {
+	// Ensure required parameters were provided
+	if c.SecurityUserName == "" {
+		return errors.New("missing SECURITY_USER_NAME")
+	}
+	if c.SecurityUserPassword == "" {
+		return errors.New("missing SECURITY_USER_PASSWORD")
+	}
+	if c.VaultToken == "" {
+		return errors.New("missing VAULT_TOKEN")
+	}
+
+	// If these values aren't perfect, we can fix them
+	if c.Port[0] != ':' {
```

Comment:
`strings.HasPrefix` would be appropriate here. As is this will panic if `len(c.Port) == 0`.

Reply:
Ah, excellent point.